xblock/runtime.py

```diff
@@ -1278,6 +1284,31 @@ def strftime(self, dtime, format):      # pylint: disable=redefined-builtin
         Locale-aware strftime, with format short-cuts.
         """
         format = self.STRFTIME_FORMATS.get(format + "_FORMAT", format)
-        if isinstance(format, unicode):
+        if six.PY2 and isinstance(format, six.text_type):
             format = format.encode("utf8")
-        return dtime.strftime(format).decode("utf8")
+        timestring = dtime.strftime(format)
+        if six.PY2:
+            timestring = timestring.decode("utf8")
+        return timestring
+
+    @property
+    def ugettext(self):
+        """
+        Dispatch to the appropriate gettext method
+        """
+        # pylint: disable=no-member
+        if six.PY3:
+            return self._translations.gettext
+        else:
+            return self._translations.ugettext
```

Comment:
I'm a bit concerned that the underlying functions change names on 2 vs 3, but our wrapper does not?  At least the docstring should make clear what this function will do.